packages/@aws-cdk/aws-cognito-identitypool/README.md

```diff
@@ -0,0 +1,366 @@
+
+A basic Identity Pool with minimal configuration has no required props, with default authenticated (user) and  
+unauthenticated (guest) roles applied to the identity pool: 
+
+```ts
+new cognito.IdentityPool(this, 'myIdentityPool');
+```
+
+By default, both the authenticated and unauthenticated roles will have no permissions attached. Grant permissions 
+to roles using the public `authenticatedRole` and `unauthenticatedRole` properties: 
+
+```ts
+const identityPool = new cognito.IdentityPool(this, 'myIdentityPool');
+const table = new dynamodb.Table(this, 'MyTable');
+
+// Grant permissions to authenticated users
+table.grantReadWriteData(identityPool.authenticatedRole);
+// Grant permissions to unauthenticated guest users
+table.grantRead(identityPool.unauthenticatedRole);
+
+//Or add policy statements straight to the role
+identityPool.authenticatedRole.addToPrincipalPolicy(new PolicyStatement({
+  effect: Effect.ALLOW,
+  actions: ['dynamodb:*'],
+  resources: ['*']
+}));
+```
+
+The default roles can also be supplied in `IdentityPoolProps`:
+
+```ts
+const stack = new Stack();
+const authenticatedRole = new Role(stack, 'authRole', {
+  assumedBy: new ServicePrincipal('service.amazonaws.com'),
+});
+const unauthenticatedRole = new Role(stack, 'unauthRole', {
+  assumedBy: new ServicePrincipal('service.amazonaws.com'),
+});
+const identityPool = new IdentityPool(stack, 'TestIdentityPoolActions', {
+  authenticatedRole, unauthenticatedRole
+});
+```
+
+### Authentication Providers
+
+Authenticated identities belong to users who are authenticated by a public login provider (Amazon Cognito user pools,  
+Login with Amazon, Sign in with Apple, Facebook, Google, SAML, or any OpenID Connect Providers) or a developer provider  
+(your own backend authentication process).
+
+[Authentication providers](https://docs.aws.amazon.com/cognito/latest/developerguide/external-identity-providers.html) can be associated with an Identity Pool by first associating them with a Cognito User Pool or by  
+associating the provider directly with the identity pool.
+
+#### User Pool Authentication Provider
+
+In order to attach a user pool to an identity pool as an authentication provider, the identity pool needs properties
+from both the user pool and the user pool client. For this reason identity pools use a `UserPoolAuthenticationProvider`
+to gather the necessary properties from the user pool constructs.
+
+```ts
+const userPool = new cognito.UserPool(this, 'Pool');
+
+new cognito.IdentityPool(this, 'myidentitypool', {
+  identityPoolName: 'myidentitypool',
+  authenticationProviders: {
+    userPool: new UserPoolAuthenticationProvider({ userPool }),
+  },
+});
+```
+
+The properties can also be passed directly to the identity pool.
+
+```ts
+new cognito.IdentityPool(this, 'myidentitypool', {
+  identityPoolName: 'myidentitypool',
+  authenticationProviders: {
+    userPool: new UserPoolAuthenticationProvider({
+      clientId: myUserPoolClientId,
```

Comment:
Is this currently implemented?

Reply:
Yes, since `UserPool` and `UserPoolClient` are separate constructs, and a `UserPool` is only useful in the context of an `IdentityPool` if it's got a client attached, I added a class to configure a `UserPool` for use with an `IdentityPool`. You can find it [here](https://github.com/smguggen/aws-cdk/blob/identity-pool/packages/%40aws-cdk/aws-cognito-identitypool/lib/identitypool-user-pool-authentication-provider.ts)

Reply:
It doesn't look like it is currently possible to pass in your own user pool client id, is it?

Reply:
Sure you can, the type that's required is `IUserPoolAuthenticationProvider`, which the `UserPoolAuthenticationProvider` class implements, but if you'd prefer to pass in your own you can pass in any object that implements `IUserPoolAuthenticationProvider`, for example:

```ts
new cognito.IdentityPool(this, 'myidentitypool', {
  identityPoolName: 'myidentitypool',
  authenticationProviders: {
    userPool: {
      clientId: myUserPoolClientId,
      identityProviders: [...myUserPoolIdentityProviders]
    },
  },
});
```
Although I just noticed that the ReadMe has that example wrong so I'm pushing a quick change to fix that.

Reply:
ahhh ok that makes sense!